### PR TITLE
Error Labeling

### PR DESCRIPTION
**Pull request by natalie-isak**
This PR creates a new module that will assign error labels to predicted detections. By comparing them to ground truth detections, the module creates an attribute called self._match_matrix, that saves the relationship between ground truth detection i and predicted detection j. 

Error types are based on this Git project: https://github.com/dbolya/tide/tree/master/tidecv:
image

Here are some examples of the matrix per image scenario:
image

### CODE REVIEW

File: python/vision_explanation_methods/error_labeling/error_labeling.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output, with an additional label column.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param task_type: The task to run.
```
Comment (imatiach-msft) [inline python/vision_explanation_methods/error_labeling/error_labeling.py:69]: I'm not sure if we need this task_type.  Maybe keep it as is for now?  I just don't know if we will really need this for other tasks.  I guess the other ones we have are much simpler right now, perhaps with image segmentation they will be more complicated again and will require this.

File: python/vision_explanation_methods/error_labeling/error_labeling.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        :type task_type: str
+        :param classes: Class names as a list of strings.
```
Comment (imatiach-msft) [inline python/vision_explanation_methods/error_labeling/error_labeling.py:71]: I don't see classes and image_mode used here - remove?

**Review by imatiach-msft — APPROVED** [review]
